### PR TITLE
add proxy option

### PR DESCRIPTION
**Pull request by jp-modernisation-gouv-fr**
For av servers installed in an environment that use a proxy, allow to use the proxy when requesting baseUrl
Signed-off-by: Jérôme Ploquin <jerome.ploquin@modernisation.gouv.fr>

### CODE REVIEW

File: src/reporting.js [hunk 1]
```diff
@@ -194,8 +195,15 @@ async function _generateReportFromDownload(console, domain, mediaId, matrixFile,
     try {
         downloadHeaders = await new Promise((resolve, reject) => {
             let responseHeaders;
+	    let connect = {url: httpUrl, encoding: null, qs: thumbnailQueryParams};
+	    const config = getConfig();
+	    const proxy = config.proxy;
+	    if (proxy != null) {
+		connect = {url: httpUrl,proxy: proxy, encoding: null, qs: thumbnailQueryParams};
```
[the remark this answers is not in the document]
Comment (dbkr) [reply src/reporting.js]: Looks like this fix didn't get included?

[the remark this answers is not in the document]
Comment (dbkr) [reply src/reporting.js]: Also, better to just say `connect.proxy = proxy;` rather than replacing the whole object unnecessarily.

**Comment by jp-modernisation-gouv-fr** [comment]
done

**Comment by michaelkaye** [comment]
I believe we may need to not use the generic HTTP_PROXY in the environment, as the call to the anti-virus server may need a different configuration to the call back to the homeserver (as the HS is in another datacenter, but the anti-virus is within the datacenter). But it's good to know that's possible.

**Pull request merged**